setup/building-es2k-stratum-deps.md

```diff
@@ -0,0 +1,221 @@
+
+- Install the ACC SDK
+
+  See [Installing the ACC SDK](../docs/guides/es2k/installing-acc-sdk.md)
+  for directions.
+
+## 3. Getting the Source Code
+
+The script to build the Stratum dependencies is in the IPDK networking-recipe
+repository.
+
+To clone the repository:
+
+```bash
+git clone --recursive https://github.com/ipdk-io/networking-recipe.git ipdk.recipe
+```
+
+You may omit the `--recursive` option if you are only interested in building
+the dependencies.
+
+You may substitute your own local directory name for `ipdk.recipe`.
+
+The build script for the Stratum dependencies is in the `setup` directory.
+
+The source code for the dependencies is not part of the distribution.
+It is downloaded by the build script.
+
+## 4. Building the Host Dependencies
+
+First, decide where to install the Host dependencies. This location (the
+"install prefix") must be specified when you configure the build.
+
+It is recommended that you *not* install the Host dependencies in `/usr` or
+`/usr/local`. It will be easier to rebuild or update the dependencies if
+their libraries are mingled with other libraries.
+
+The `setup` directory includes a helper script (`make-host-deps.sh`) that
+can be used to build the Host dependencies.
+
+- The `--help` (`-h`) option lists the parameters the helper script supports
+
+- The `--dry-run` (`-n`) option displays the parameter values without
+  running CMake
+
+The script normally does a minimal build, containing just the components
+needed for cross-compilation. Specify the `--full` parameter if you want
+to build all the libraries.
+
+> **Note:** The Host and Target build environments are mutually incompatible.
+  You must ensure that the [target build environment variables](#5-defining-the-target-build-environment)
+  are undefined before you build the Host dependencies.
+
+### User build
+
+To install the dependencies in a user directory:
+
+```bash
+./make-host-deps.sh --prefix=PREFIX
+```
+
+PREFIX might something like `~/hostdeps`.
+
+The source files will be downloaded and built, and the results will be
+installed in the specified directory.
+
+### System build
+
+To install the Host dependencies in a system directory, log in as `root`
+or build from an account that has `sudo` privilege.
+
+```bash
+./make-host-deps.sh --prefix=PREFIX --sudo
+```
+
+PREFIX might be something like `/opt/ipdk/x86deps`.
+
+The script only uses `sudo` when installing libraries. Omit the parameter
+if you are running as `root`.
+
+## 5. Defining the Target Build Environment
+
+In order to cross-compile for the ACC, you will need to define a number
+of environment variables. This is typically done by putting the bash
+commands in a file (e.g. `es2k-setup.env`) and using the `source` command
+to execute it. We recommend removing execute permission from the file
+(`chmod a-x setup.env`) to remind yourself to source it, not run it.
+
+For example:
+
+```bash
+ACC_SDK=<acc-sdk-directory>
```

Comment:
Is this a pre-defined environment variable that we use? Im wondering if there might be confusion with the _IPU SDK_ that users will be using going forward

Reply:
ACC_SDK is local to this script, as are P4CPBASE, AARCH64, and SYSROOT.

ACC_SDK and P4CPBASE are discrete variables that the user sets directly. I do this so they don't have to edit multiple occurrences or change something that is part of a longer string. If this were a template file, the values would be marked for replacement (@ACC_SDK@ and @P4CPBASE@) at configuration time.

AARCH64 and SYSROOT are local variables that represent the two directory trees we are mapping.

None of these variables is `export`ed. I debated making them explicitly `local` and possibly `unset`ting them at the end of the script, but that seemed a tad excessive.

The variable that we are publishing are the ones documented in lines 154-165.

Reply:
I've gone through several iterations of this. First, I added a bunch of comments to the sample configuration script. This seemed excessive, especially for ten lines of bash in a document, so I removed the comments and added 28 lines explanatory text the document. I've just push a change that adds three comments and two blank lines to the document, and moves one of the existing lines. I think this will do for now.

I intend to extract this section and make it a separate document (it applies to cross-compiling P4CP as well), but I decide it was better to make the documentation available in this form, right now, instead of waiting for the P4CP document to be complete.